tests/test_status.py

```diff
@@ -0,0 +1,87 @@
+import pytest
+from flask import jsonify, Flask
+from datetime import datetime, timezone, timedelta
+import json
+from contextlib import contextmanager
+from flask import appcontext_pushed, g
+
+app = Flask(__name__)
+
+@app.route('/api/deployments')
+def deployments_mock():
+    return jsonify(_generate_status_json())
+
+@pytest.fixture
+def client():
+    app.config["TESTING"] = True
+    app.config["DEBUG"] = False
+    with app.test_client() as client:
+        yield client
+
+def test_deployments_api(client):
+    with app.app_context():
```

Comment:
I don't know if this many levels of nesting is strictly needed for the tests, but I had difficulty getting things working otherwise with the mock request.  Any suggested revisions would be welcome here.